modules/core/shared/src/main/scala/lucuma/core/math/ProperMotion.scala

```diff
@@ -32,36 +32,40 @@ final case class ProperMotion(
   dec: ProperMotion.AngularVelocityComponent[VelocityAxis.Dec]
 ) {
   // Return the ra/dec components in radians, first converting to degrees/y
-  def toRadians: (Double, Double) =
+  inline def toRadians: (Double, Double) =
     (ra.toRadians, dec.toRadians)
 
 }
 
 object ProperMotion extends ProperMotionOptics {
-  final case class AngularVelocityComponent[A](μasy: Quantity[Long, MicroArcSecondPerYear]) {
+  opaque type AngularVelocityComponent[A] = Quantity[Long, MicroArcSecondPerYear]
```

Comment:
Here's an idea, since we are revising the model: `A` seems to be just a tag.

What if we use a `NewType` for `AngularVelocityComponent` (without `A`) and then we use tagged types in `ProperMotion`?

The. new type can be just `AngularVelocity`, and it would look something like this: `case class ProperMotion(ra: AngularVelocity Of VelocityAxis.RA, dec: AngularVelocity Of VelocityAxis.Dec)`.

Reply:
I'm not so sure what the `A` is doing there and in fact it is not limited to RA/Dec thus you could tag anything.
Could we have two newtypes
```object AngularVelocityRA extends NewType[Quantity[Long, MicroArc...]]]
object AngularVelocityDec extends NewType[Quantity[Long, MicroArc...]]]```

is the `Of` mechanism using Shapeless tags?

Reply:
Isn't `Of` or a shapeless tag essentially what an `opaque type` would do?

Reply:
`A Of T` is actually an opaque type over `T` that is also a subtype of `T`:

``` scala
opaque infix type Of[+T, U] <: T = T
```

You can turn an `a: A` into an `A Of T` by calling `tag[T](a)`.

The difference with a non-generic opaque type is that the type `A` is exposed.

Also the tag `T` is an independent type that can be propagated to tag other types. This is not necessary here, but it's what is done to relate tagged `Units` and `Measure`.

If we use `Of` here (or anywhere else), we should move it out of the `math.dimensional` package and into `util` maybe?